Tests/SwiftDocCTests/Infrastructure/ExternalReferenceResolverTests.swift

```diff
@@ -1155,4 +1155,232 @@ class ExternalReferenceResolverTests: XCTestCase {
+            """)
+        ) { symbol in
+            let expectedMarkupsWithResolvedLink = [
+                "For more information about this value, see <doc://com.external.testbundle/externally/resolved/path> instead.",
+                "For more information about this other value, see <doc://com.external.testbundle/externally/resolved/path> instead."
+            ]
+            let httpParametersSection = try XCTUnwrap(symbol.httpParametersSection)
+            XCTAssertEqual(httpParametersSection.parameters.count, 2)
+            try httpParametersSection.parameters.forEach {
+                try assertMarkupHasResolvedLink(markupsWithResolvedLink: expectedMarkupsWithResolvedLink, actualMarkup: $0.contents)
+            }
+        }
+    }
+
+    func testHTTPBodyWithExternalLink() throws {
+        try withExampleRESTAPIDocumentation(
+            TextFile(name: "Some-Data.md", utf8Content: """
+            # ``Some-Data``
+
+            Retrieve some data from our database.
+
+            - HTTPBody: Read this instead: <doc://com.external.testbundle/something-else>.
+
+            """)
+        ) { symbol in
+            let expectedMarkupsWithResolvedLink = "Read this instead: <doc://com.external.testbundle/externally/resolved/path>."
+            let httpBodySection = try XCTUnwrap(symbol.httpBodySection)
+            try assertMarkupHasResolvedLink(markupsWithResolvedLink: [expectedMarkupsWithResolvedLink], actualMarkup: httpBodySection.body.contents)
+        }
+    }
+
+    func testHTTPBodyParametersWithExternalLink() throws {
+        try withExampleRESTAPIDocumentation(
+            TextFile(name: "Some-Data.md", utf8Content: """
+            # ``Some-Data``
+
+            Retrieve some data from our database.
+
+            - HTTPBodyParameters:
+                - artist: Read more about artists: <doc://com.external.testbundle/artists>.
+                - userName: Read more about user names: <doc://com.external.testbundle/user-names>.
+
+            """)
+        ) { symbol in
+            let expectedMarkupsWithResolvedLink = [
+                "Read more about artists: <doc://com.external.testbundle/externally/resolved/path>.",
+                "Read more about user names: <doc://com.external.testbundle/externally/resolved/path>."
+            ]
+            let httpBodySection = try XCTUnwrap(symbol.httpBodySection)
+            XCTAssertEqual(httpBodySection.body.parameters.count, 2)
+            try httpBodySection.body.parameters.forEach {
+                try assertMarkupHasResolvedLink(markupsWithResolvedLink: expectedMarkupsWithResolvedLink, actualMarkup: $0.contents)
+            }
+        }
+    }
+
+    func testHTTPResponsesWithExternalLink() throws {
+        try withExampleRESTAPIDocumentation(
+            TextFile(name: "Some-Data.md", utf8Content: """
+            # ``Some-Data``
+
+            Retrieve some data from our database.
+
+            - HTTPResponses:
```

Comment:
Is it / should it be valid to document http responses, parameters, body, etc. for a symbol that doesn't have a `restResponses` `restParameters`, `restBody`,`restBody/parameters`, etc. in the symbol graph file?

If the test data isn't realistic and we add more validation in the future then these tests will fail, without detecting a real bug, and it's the person who adds the validation who has to go back and fix old test data.

Reply:
Agreed. I found some realistic symbol graphs files that are present under the `Tests/SwiftDocCTests/Test Bundles` directory, and refactored these new unit test to use those instead, while addressing your other comments above. I think the tests are easier to read now, and hopefully will be more reliable.

Reply:
This is true for the tests related to the HTTP/Rest endpoint symbol, and for the dictionary keys test. I also added a test for normal symbol parameters which was missing earlier. For the symbol parameters I create a fake symbol graph in memory like the other unit tests in this file do.